src/openloco/ui.cpp

```diff
@@ -776,22 +780,96 @@ namespace openloco::ui
         SDL_ShowSimpleMessageBox(SDL_MESSAGEBOX_BUTTON_RETURNKEY_DEFAULT, title.c_str(), message.c_str(), window);
     }
 
-    bool updateSDLResolution()
+    bool setDisplayMode(config::screen_mode mode, int16_t width, int16_t height)
     {
-        SDL_DisplayMode mode;
-        SDL_GetWindowDisplayMode(window, &mode);
+        SDL_DisplayMode DisplayMode;
+        SDL_DisplayMode DesktopDisplayMode;
+        Resolution DesktopResolution;
+        Resolution FallbackResolution;
+        int32_t DisplayIndex = SDL_GetWindowDisplayIndex(window);
+        SDL_GetDesktopDisplayMode(DisplayIndex, &DesktopDisplayMode);
+        SDL_GetDesktopDisplayMode(DisplayIndex, &DisplayMode);
+        DesktopResolution = { DesktopDisplayMode.w, DesktopDisplayMode.h };
+
+        auto& old_config = config::get();
+        auto& config = config::get_new();
 
-        auto& cfg = config::get();
-        mode.h = cfg.resolution_height;
-        mode.w = cfg.resolution_width;
+        config.display.mode = mode;
+
+        auto flags = 0;
+        switch (mode)
+        {
+            case config::screen_mode::window:
+                FallbackResolution = { config.display.window_resolution.width, config.display.window_resolution.height };
```

Comment:
This is copying the values into a `Resolution` in several places. Instead, I suggest you do this in two steps:
1. Assign a reference to the relevant configuration struct (`config::resolution_t type`) as `fallbackResolution`.
2. Convert them in one place.

It would be better if the logic for this is in a separate function, e.g. `getDisplayResolutionByMode(...)`. This makes it easier to understand the function at a glance. It is a little too big now.

(We don't _have_ to call this potential `getDisplayResolutionByMode` in any other function. The compiler will inline it if it deems it more efficient.)